### PR TITLE
Meganav-rebase

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Final PR for the new meganav release

## QA

- Check that the appearance, functionality and content are the same in [this demo](https://ubuntu-com-12939.demos.haus/) as they are in [the original demo](https://ubuntu-com-12252.demos.haus/)
- Check the code diffs and see that that no new changes have been introduced (unless they are fixes)

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-4297

### CODE REVIEW

**Comment by petesfrench** [comment]
@ClementChaumel can you take a look at this please?

**Comment by petesfrench** [comment]
these changes to be applied to this PR https://github.com/canonical/ubuntu.com/pull/13020

**Comment by petesfrench** [comment]
@carkod Did you say you would take a look at this?

**Comment by carkod** [comment]
I think you have some conflicts, and maybe @juanruitina wants to take a look?

I'll need some time to review it all.

File: build.js [hunk 1]
```diff
@@ -26,7 +26,6 @@ let entries = {
   "certified-search-results": "./static/js/src/certified-search-results.js",
   openstackDeploymentChart: "./static/js/src/openstack-deployment-chart.js",
   blender: "./static/js/src/advantage/subscribe/blender/app.tsx",
-  tabbedContent: "./static/js/src/tabbed-content.js",
```
Comment (carkod) [inline build.js:29]: Why is this being removed? Many pages have this dependency?

Comment (petesfrench) [reply build.js:29]: It is needed in the navigation, so I made it globally available via main.js. If you check some other pages they should still be working fine.

File: static/js/src/main.js [hunk 2]
```diff
@@ -10,3 +10,4 @@ import "./navigation.js";
 import "./mobile-footer-navigation.js";
 import "./smart-quotes.js";
 import "./intlTelInput.js";
+import "./tabbed-content.js";
```
Comment (carkod) [inline static/js/src/main.js:13]: ok, I see it. Thanks

File: static/js/src/navigation.js [hunk 3]
```diff
@@ -1,91 +1,209 @@
...
+  const isActive = clickedDropdown.classList.contains("is-active");
+
+  dropdownWindow.classList.remove("slide-animation");
+  dropdownWindowOverlay.classList.remove("fade-animation");
+  toggleNavMenu(clickedDropdown, !isActive);
+}
+
+function toggleNavMenu(dropdown, show) {
+  const dropdownContent = document.getElementById(dropdown.id + "-content");
+  const dropdownContentMobile = document.getElementById(
```
Comment (carkod) [inline static/js/src/navigation.js:49]: Might be worth moving to the top? As this seems to be used in several places (line 29)

File: static/sass/_pattern_navigation-dropdown.scss [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    }
+
+    [class*="p-button"] {
+      background: transparent;
+      margin: 0;
+    }
+  }
+
+  .dropdown-window__side-panel {
+    margin-top: 1rem;
```
Comment (carkod) [inline static/sass/_pattern_navigation-dropdown.scss:152]: I believe you can replace all the `1rem` with `$spv-large` https://vanillaframework.io/docs/migration-guide-to-v3

File: static/sass/styles.scss [hunk 5]
```diff
@@ -527,6 +530,10 @@ summary {
       color: inherit;
     }
   }
+
+  .is-muted {
```
Comment (carkod) [inline static/sass/styles.scss:534]: Have you tried with `u-text--muted`? https://vanillaframework.io/docs/base/typography#muted-text

**Comment by petesfrench** [comment]
@carkod thanks for the review, but at this point most of these changes aren't relevant anymore. As most of this is updated in slide-nav PR. The QA needed here is to ensure nothing was lost during the rebase and that everything is in the same state as it was before the rebase.

**Review by carkod — DISMISSED** [review]

**Comment by juanruitina** [comment]
I just made some content changes in the YAML file to incorporate feedback from Gloria and Martin.

One little bug: when logged out, the chevron shouldn't be visible in the "Sign in" link on the top right.

Screenshot 2023-11-08 at 12 50 27

**Comment by juanruitina** [comment]
Some of the minor bugs I've noticed/Pete had identified, handle them as you consider:

### Search field placeholder

The search field placeholder is currently unreadable (white). We had this issue elsewhere if I remember right. Maybe it has been/should be addressed on Vanilla, can you check? I see it doesn't happen [in the Vanilla website itself](https://vanillaframework.io/).

Screenshot 2023-12-05 at 11 35 08

### Little visual bug on mobile

I noticed a visual bug persists on mobile: see the darker rectangle.

Screenshot 2023-12-05 at 11 49 21

But I think I found the culprit! Disabling the box shadow CSS for `.p-navigation--sliding .p-navigation__dropdown::before` seems to fix it, can you check?

### Keyboard navigation: focus after last link in dropdown

When we open, e.g., "Products", and then select a tab on the second level, e.g. "Ubuntu OS", and then tab through all the links until the end, hitting tab after the last link closes the dropdown and moves the focus to the next item on the top navigation (in this case "Use cases"). This is correct and the [intended behaviour](https://warthogs.atlassian.net/browse/WD-7516?focusedCommentId=349710).

However, when we open, e.g., "Products" and then tab through all the links _without_ selecting a tab on the second level, hitting tab after the last link moves the focus past the whole navigation into the content, with the navigation remains open. The behaviour here should be exactly the same as above: it should close the dropdown and move focus to the next top level item, "Use cases" in this case.

### Keyboard navigation on mobile

Also on keyboard navigation, Pete mentioned there was up a bug when tabbing on secondary links on mobile. I see indeed that focus seems to move to out of screen elements when tabbing through quick links there. E.g., under Products > Ubuntu OS, focus seems lost once you move past the Multipass link, instead of jumping to the first quick link (Upgrade Ubuntu).

**Comment by juanruitina** [comment]
Two other visual bugs:

- The padding of the menu button needs to be adjusted in bubbles on mobile, should match that of "Canonical Ubuntu" next to it. (Please check that whatever you do here doesn't affect the button on the home, which has different padding)

ubuntu-com-12939 demos haus_pro (1)

- There's an alignment issue between navigation and the third column in rebranded pages. This looks OK in the [/data bubble](https://canonical.com/data) and [microk8s.io](https://microk8s.io/). In the example below, "Products" and "Your subscriptions" should align with "Ubuntu Pro" in the hero; see how it does in the Data bubble below that.

ubuntu-com-12939 demos haus_pro
canonical com_data

**Comment by akbarkz** [comment]
@petesfrench I tested it on iPhone 11 and Huawei P20, so both on iOS and Android. 
Click on Menu and choose Use cases. This will open a long menu. Scroll all the way down.
- on iOS it stops on Industrial, so that Robotics and Telco are left invisible below. You can hold and scroll down to expose both Robotics and Telco, but as soon as you release your finger they will hide again.
- on Android it stops on Robotics and there is no way to see Telco that is hidden below. Holding and scrolling down doesn't do anything. 
This is a bug that doesn't allow to click on items that are overflown.

File: meganav.yaml [hunk 6]
```diff
@@ -0,0 +1,951 @@
...
+              url: https://help.ubuntu.com/community/ReportingBugs
+            - title: Software development
+              description: Create and maintain software
+              url: https://ubuntu.com/community/contribute/software-development
+            - title: Improve documentation
+              description: Help write the docs
+              url: https://wiki.ubuntu.com/DocumentationTeam/SystemDocumentation
+            - title: Art and design
+              description: Contribute your artwork
+              url: https://ubuntu.com/community/contribute/art-design
```
Comment (anthonydillon) [inline meganav.yaml:682]: The ubuntu.com links here should be relative.

File: meganav.yaml [hunk 7]
```diff
@@ -0,0 +1,951 @@
...
+get_ubuntu:
+  side_nav_sections:
+    - title: Desktop
+      primary_links:
+        - links:
+            - title: Ubuntu Desktop
+              description: Free and reliable desktop OS
+              url: /download/desktop
+              secondary_cta_title: Download 22.04 LTS
+              secondary_cta_url: /download/desktop/thank-you?version=22.04.3&architecture=amd64
```
Comment (anthonydillon) [inline meganav.yaml:767]: Shouldn't these values come from the releases.yaml?

**Comment by anthonydillon** [comment]
Added a few comments in the nav file. What is holding up this from being released at this point?

File: meganav.yaml (hunk 6 above)
Comment (juanruitina) [reply meganav.yaml]: Fixed! Thank you Ant

File: meganav.yaml (hunk 7 above)
Comment (petesfrench) [reply meganav.yaml:767]: Yes, good point. I have made an issue in our backlog to address this later: https://warthogs.atlassian.net/browse/WD-8596

**Comment by petesfrench** [comment]
> Added a few comments in the nav file. What is holding up this from being released at this point?

I just need to rebase the feature branch and we are good to go. I think the change of pulling download data from the releases.yaml can be dealt with post-merge.

**Comment by anthonydillon** [comment]
Screen wide (1281px) looks broken.

**Review by akbarkz — APPROVED** [review]

**Pull request merged**